server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/util/ReplacingServiceImpl.java

```diff
@@ -120,4 +209,30 @@ private String replace(@NonNull String text, @NonNull Pair<String, String> patte
     }
     return result;
   }
+
+  private Function<String, Boolean> checkContainWord(String check) {
+    return text ->
+        Arrays.stream(text.toUpperCase().split("\b")).anyMatch(txt -> txt.equalsIgnoreCase(check));
+  }
+
+  @NonNull
+  private ProcessedSearchClause getProcessedSearchClause(@NonNull String clause) {
+    SearchPattern searchPattern = SearchPattern.EXACT;
+    Matcher matcher = LEAD_OR_TRAIL_CLAUSE.matcher(clause);
+    if (matcher.matches()) {
+      if (matcher.group(1).trim().equalsIgnoreCase("LEADING"))
+        searchPattern = SearchPattern.STARTS_WITH;
+      else if (matcher.group(1).trim().equalsIgnoreCase("TRAILING"))
```

Comment:
You don't need to trim the group if you change your pattern to `"\\s*(LEADING|TRAILING).*"` (without space after | sign).

Reply:
done